### PR TITLE
Ensure examples aren't double encoded in output

### PR DESCRIPTION
**Pull request by WarpSpideR**
Corresponding fix for issue #99.

### CODE REVIEW

File: src/Swashbuckle.AspNetCore.Filters/Examples/OpenApiRawString.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
+﻿using Microsoft.OpenApi;
+using Microsoft.OpenApi.Any;
+using Microsoft.OpenApi.Writers;
+
+namespace Swashbuckle.AspNetCore.Filters
+{
+    /// <summary>
+    /// Represents a raw value that should not be encoded
+    /// </summary>
+    public class OpenApiRawString : IOpenApiAny, IOpenApiPrimitive
```
Comment (mattfrear) [inline src/Swashbuckle.AspNetCore.Filters/Examples/OpenApiRawString.cs]: Perhaps make it an internal class

File: src/Swashbuckle.AspNetCore.Filters/Examples/RequestExample.cs [hunk 2]
```diff
@@ -41,17 +41,20 @@ internal class RequestExample
...
             if (operation.RequestBody.Content.Keys.Any(k => k.Contains("xml")))
             {
-                xmlExample = new OpenApiString(example.XmlSerialize());
+                xmlExample = new OpenApiRawString(example.XmlSerialize());
             }
 
             foreach (var content in operation.RequestBody.Content)
             {
-                content.Value.Example = content.Key.Contains("xml") ? xmlExample : jsonExample;
+                if (content.Key.Contains("xml"))
```
Comment (mattfrear) [inline src/Swashbuckle.AspNetCore.Filters/Examples/RequestExample.cs]: Please add { } around the if to keep code style consistant

File: src/Swashbuckle.AspNetCore.Filters/Examples/ResponseExample.cs [hunk 3]
```diff
@@ -40,17 +40,20 @@ internal class ResponseExample
...
                 if (response.Value.Content.Keys.Any(k => k.Contains("xml")))
                 {
-                    xmlExample = new OpenApiString(example.XmlSerialize());
+                    xmlExample = new OpenApiRawString(example.XmlSerialize());
                 }
 
                 foreach (var content in response.Value.Content)
                 {
-                    content.Value.Example = content.Key.Contains("xml") ? xmlExample : jsonExample;
+                    if (content.Key.Contains("xml"))
```
Comment (mattfrear) [inline src/Swashbuckle.AspNetCore.Filters/Examples/ResponseExample.cs]: Please add { }

File: src/Swashbuckle.AspNetCore.Filters/Swashbuckle.AspNetCore.Filters.csproj [hunk 4]
```diff
@@ -27,6 +27,7 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="2.0.0" />
+    <PackageReference Include="Microsoft.OpenApi" Version="1.1.3" />
```
Comment (mattfrear) [inline src/Swashbuckle.AspNetCore.Filters/Swashbuckle.AspNetCore.Filters.csproj]: Just wondering why you've added a PackageReference to this, I guess it is added by Swashbuckle already?

Comment (WarpSpideR) [reply src/Swashbuckle.AspNetCore.Filters/Swashbuckle.AspNetCore.Filters.csproj]: Yes, it was added by Swashbuckle already but that version isn't compatible with the `OpenApiRawString` class. There've got a strange way of handling `OpenApiPrimatives` that causes problems when casting using implicit operators. They've added better handling of polymorphism in the newer version which I'm hoping will be added to Swashbuckle in the future and we can remove this reference.

File: src/Swashbuckle.AspNetCore.Filters/Examples/OpenApiRawString.cs (hunk 1 above)
Comment (WarpSpideR) [reply src/Swashbuckle.AspNetCore.Filters/Examples/OpenApiRawString.cs]: Ahh yes, my bad!

File: src/Swashbuckle.AspNetCore.Filters/Examples/RequestExample.cs (hunk 2 above)
Comment (WarpSpideR) [reply src/Swashbuckle.AspNetCore.Filters/Examples/RequestExample.cs]: Consider it done.

**Comment by mattfrear** [comment]
Sorry for my tardiness on this... I flew from the UK to NZ on Wednesday so I lost a couple of days travelling and I am still jet lagged, so I have no time in the evenings. I'll get to it soon.

**Comment by mattfrear** [comment]
Houston, we have a problem. If I run the `WebApi2.0-Swashbuckle5` from the solution and then look at the swagger.json, it is invalid, because of the XML examples:
this pr:
```
              "application/xml": {
                "schema": {
                  "$ref": "#/components/schemas/PersonResponse"
                },
                "example": <PersonResponse xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">

  <Id>123</Id>

  <Title>Dr</Title>

  <FirstName>Hank</FirstName>

  <LastName>Thomas</LastName>

  <Age>27</Age>

  <Income xsi:nil="true" />

</PersonResponse>
              },
```
master branch:
```
              "application/xml": {
                "schema": {
                  "$ref": "#/components/schemas/PersonResponse"
                },
                "example": "<PersonResponse xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xmlns:xsd=\"http://www.w3.org/2001/XMLSchema\">\r\n  <Id>123</Id>\r\n  <Title>Dr</Title>\r\n  <FirstName>Hank</FirstName>\r\n  <LastName>Thomas</LastName>\r\n  <Age>27</Age>\r\n  <Income xsi:nil=\"true\" />\r\n</PersonResponse>"
              },
```

I still don't have much spare time but I can look at it when I get a chance, maybe some time in the next week.

**Comment by WarpSpideR** [comment]
No worries, I'm just back from holiday as well so we'll call it even!

Ahh, that's totally my bad, I should have tested that part as well! I'll try and find some time, probably tomorrow to find a fix for this!

**Comment by WarpSpideR** [comment]
Sorry for the delay, things have been a bit hectic over here but we got there in the end. 

I've added a fix for XML generation and some tests to ensure XML is surrounded by quotation marks and that the JSON output is not.

**Comment by mattfrear** [comment]
Nice one, thanks again! I'll release it to NuGet now.

**Pull request merged**